base/ltcounts.dtx

```diff
@@ -56,14 +56,17 @@
 %</driver>
 % \fi
 %
+% \providecommand\env[1]{\texttt{#1}}
 %
 % \changes{v1.0c}{1994/03/29}
 %     {Create file from parts of ltmiscen and ltherest.}
 % \changes{v1.1a}{1994/05/19}{Extracted file from ltcntlen.}
 % \changes{v1.1d}{1996/04/23}{Documentation improvements}
 %
 % \section{Counters and Lengths}
-% Commands for defining and using counters. This file defines:
+% Commands for defining and using counters.
+% This file defines the following commands. In each case \marg{counter} may be
+% |*| denoting the current counter as set by a previous |\refstepcounter|.
```

Comment:
That's a bad idea for `\counterwithin`/`\counterwithout`, isn't it?

Reply:
@Skillmon  er probably

```

\show\cl@subsection
\counterwithin{equation}{*}
\show\cl@subsection
\expandafter\show\csname cl@*\endcsname

\show\cl@equation
\counterwithin{*}{equation}
\show\cl@equation

```

but what would you have it do in either case, be an error or act as if the current counter name had been used at that point?

Reply:
or not be an error, and do what it does now, which is put the `*` in the reset list so the reset works on whatever the current counter is at that time, which is a bit weird but sort of what the user asks for.

Reply:
I think I would change `\@ifbothcounters` and generate an error "*-alias for current counter not supported", or you could expand there any * to the current counter and then do what it currently does, but I think if anything it should generate an error (or just a documentation that with those you get what you get :-))

Reply:
I agree, but that could be handled by just documenting it, rather then runtime processing for something nobody (sane) would even try. But I leave this up to @davidcarlisle

Reply:
could do something like

```
\def\@ifbothcounters#1#2#3{%
  \begingroup\let\c@*\@undefined
  \@ifundefined{c@#1}{\endgroup\@nocounterr{#1}}%
    {% else counter is defined
     \@ifundefined{c@#2}{\endgroup\@nocounterr{#2}}%
       {\endgroup% else both counter and within  are defined
        #3}}}
```

But I'm tempted just to adjust the documentation and say don't use `*` it's not like the `\refstepcounter` case which puts you in a loop so really need a guard, here it just does what it does.

Reply:
either or would be fine imho, but I think  you can move the \endgroup to the outside since counter defs are fully global.

Reply:
ah it's good someone knows this code:-) OK  I'll do that, it looks less weird.

Reply:
hopefully resolved by f9f49c376

Reply:
maybe one should make a remark somewhere (ltnews41?) that the * convention is not allowed on all commands.

Reply:
@FrankMittelbach  added to ltnews ddd622501